### PR TITLE
Update to be able to track both DaemonSets and Deployments

### PR DESCRIPTION
**Pull request by danwinship**
OVN uses a deployment, so we'll need to be able to track a mix of Deployments and DaemonSets now.

(In fact, failing to track the Deployment doesn't actually affect anything with OVN, because the node DaemonSet can't become ready until after the master Deployment is ready anyway. But whatever.)

Not tested yet because we appear to be having AWS account issue, but you can still review the concept...

### CODE REVIEW

File: pkg/controller/networkconfig/networkconfig_controller.go [hunk 1]
```diff
@@ -185,15 +189,24 @@ func (r *ReconcileNetworkConfig) Reconcile(request reconcile.Request) (reconcile
...
 			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
+			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		}
 	}
-	r.daemonSetReconciler.SetDaemonSets(daemonSets)
+	r.status.SetDaemonSets(daemonSets)
+	r.status.SetDeployments(deployments)
+
+	allResources := []types.NamespacedName{}
```
Comment (squeed) [inline pkg/controller/networkconfig/networkconfig_controller.go]: I realize this is a bit pathological, but you could totally have a NamespacedName collision between daemonsets and deployments.

Comment (squeed) [reply pkg/controller/networkconfig/networkconfig_controller.go]: Especially if you moved from one Kind to the other.

Comment (squeed) [reply pkg/controller/networkconfig/networkconfig_controller.go]: Ahh, now I see why that doesn't really matter.

**Comment by squeed** [comment]
Seems reasonable to me.

In theory you could move the namespace checks to the Watch predicate, but that seems more error-prone - and certainly doesn't get any extra efficiency.

**Comment by danwinship** [comment]
/hold
not sure why the bot removed my WIP label... but this is still not actually tested

File: pkg/controller/networkconfig/networkconfig_controller.go (hunk 1 above)
Comment (danwinship) [reply pkg/controller/networkconfig/networkconfig_controller.go]: I started working on a controller-runtime PR to make it include a `GroupVersionKind` in `reconcile.Request`. But as you figured out, it doesn't actually matter here.

**Comment by danwinship** [comment]
/hold
not needed for 4.0

**Comment by danwinship** [comment]
/hold cancel
the rewrites here make fixing some of the other status stuff easier

**Pull request merged**